### PR TITLE
[ADP-3272] Use consistent terminology for values and fields of `PartialTx`.

### PR DESCRIPTION
**Pull request by jonathanknowles**
## Issue

ADP-3272

## Summary

This PR:
- renames the `inputs` field of a `PartialTx` to `extraUTxO`.
- adjusts usages of `PartialTx` so that field names are used consistently.

## Details

The updated name `extraUTxO` better reflects the fact that this field:
- represents an **_extra source of UTxOs_** that can be referenced **_by_** the inputs of a partial transaction;
- is **_not_** a set of inputs, and should **_not_** be confused with the partial transaction inputs themselves.

In addition, this PR adjusts various places that pattern match on `PartialTx` values to use [named field puns](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/record_puns.html) instead of positional pattern matching. This approach helps to maintain consistency of terminology -- if later on we rename any of the fields of a `PartialTx`, the compiler will remind us to update these pattern matches.

### CODE REVIEW

File: lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs [hunk 1]
```diff
@@ -426,8 +426,7 @@ deriving instance IsRecentEra era => Show (ErrBalanceTx era)
 -- even though they are in an "unordered" set.
 data PartialTx era = PartialTx
     { tx :: Tx era
-    , inputs :: UTxO era
-      -- ^ NOTE: Can we rename this to something better? Perhaps 'extraUTxO'?
+    , extraUTxO :: UTxO era
```
Comment (jonathanknowles) [inline lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs:429]: > NOTE: Can we rename this to something better? Perhaps `extraUTxO`?

Yes we can! 😅

File: lib/balance-tx/test/spec/Internal/Cardano/Write/Tx/BalanceSpec.hs [hunk 2]
```diff
@@ -2688,19 +2690,13 @@ instance forall era. IsRecentEra era => Arbitrary (PartialTx era) where
             (fromCardanoApiUTxO inputUTxO)
             (redeemers)
             mempty
-    shrink (PartialTx tx inputUTxO redeemers timelockKeyWitnessCounts) =
-        [ PartialTx tx inputUTxO' redeemers timelockKeyWitnessCounts
-        | inputUTxO' <- shrinkInputResolution inputUTxO
+    shrink partialTx@PartialTx {tx, extraUTxO} =
+        [ partialTx {extraUTxO = extraUTxO'}
+        | extraUTxO' <- shrinkInputResolution extraUTxO
```
Comment (jonathanknowles) [inline lib/balance-tx/test/spec/Internal/Cardano/Write/Tx/BalanceSpec.hs:2694]: Here we avoid pattern matching on fields of `PartialTx` that are unaffected by shrinking.

File: lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs [hunk 3]
```diff
@@ -637,14 +637,14 @@ balanceTransactionWithSelectionStrategyAndNoZeroAdaAdjustment
     genChange
     s
     selectionStrategy
-    ptx@(PartialTx partialTx inputUTxO redeemers timelockKeyWitnessCounts)
+    partialTx@PartialTx {extraUTxO, redeemers, timelockKeyWitnessCounts}
     = do
-    guardExistingCollateral partialTx
```
Comment (Anviking) [inline lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs:642]: The explicit arguments perhaps had some usefulness for readability?

**Review by paweljakubas — APPROVED** [review]
lgtm!

**Review by Anviking — APPROVED** [review]
Thanks!

File: lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs (hunk 3 above)
Comment (jonathanknowles) [reply lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs:642]: > The explicit arguments perhaps had some usefulness for readability?

I can empathise with this argument! However I think there were a couple of drawbacks:
- We were somewhat inconsistent about the choice between implicit and explicit arguments:
    - Example: `guardWalletUTxOConsistency` compares two UTxO sets. Before this change, one of these UTxO sets was **_explicit_**, and the other **_implicit_**.
- Explicit arguments created additional aliases for values that were already brought into scope by the enclosing function. By making these arguments implicit, we avoid the cost of choosing names that don't collide with the existing names.

**Pull request merged**